jax/_src/lax/lax.py

```diff
@@ -2907,11 +2910,13 @@ def _dot_general_lower(ctx, lhs, rhs, *, dimension_numbers,
                                core.ShapedArray(rhs_aval.shape, aval_out.dtype))
         lhs_dtype = rhs_dtype = aval_out.dtype
     else:  # cpu and gpu
-      lhs = mlir.convert_hlo(ctx, lhs, lhs_aval,
-                             core.ShapedArray(lhs_aval.shape, aval_out.dtype))
-      rhs = mlir.convert_hlo(ctx, rhs, rhs_aval,
-                             core.ShapedArray(rhs_aval.shape, aval_out.dtype))
-      lhs_dtype = rhs_dtype = aval_out.dtype
+      # Do not convert mixed fp8 types to output type.
```

Comment:
My main concern here is: do we have a test case that exercises this? I'm mostly worried about possibly breaking CPU, since this path is taken for CPUs also.

Reply:
Just added tests.